clusterBuildStrategy/buildah/README.md

```diff
@@ -0,0 +1,103 @@
+# Buildah ClusterBuildStrategy
+The `buildah` ClusterBuildStrategy uses [buildah](https://github.com/containers/buildah) to build and push a container image, out of a `Dockerfile`. The `Dockerfile` should be specified using a parameter in the `Build` resource.
+
+## Getting Started with Buildah ClusterBuildStrategy
+
+### Pre-requisites:
+- The `Builds for Red Hat OpenShift` Operator is available in the OpenShift cluster.
+- The `buildah` clusterBuildStrategy is already installed. This could be checked by following command:
+```
+$ oc get cbs | grep buildah
+```
+If the buildStrategy is not available, [Install ClusterBuildStrategies](../../README.md) first.
+
+### Steps:
+
+1. Create a builds object. 
+- `source` refers to the location where the source code is.
+- `strategy` references the BuildStrategy to use to build the container.
+- `paramValues` is a list of key/value that could be used to set strategy parameters.
+    - Here, we pass the `dockerfile` parameter to specify the Dockerfile's location required to build the output image.
+- `output` refers to the location where the built image would be pushed.
+    - Here, it is expected to be pushed to the OpenShift cluster’s internal registry.
+    - `buildah-example` is the name of the current project.
+```
+$ cat <<EOF | oc create -f -
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: buildah-golang-build
+spec:
+  source:
+    git: 
+      url: https://github.com/shipwright-io/sample-go
+    contextDir: docker-build
+  strategy:
+    name: buildah-strategy-managed-push
+    kind: ClusterBuildStrategy
+  paramValues:
+  - name: dockerfile
+    value: Dockerfile
+  output:
+    image: image-registry.openshift-image-registry.svc:5000/buildah-example/sample-go-app
+EOF
+```
+
+2. Create a buildRun object. 
+- `spec.build.name` refers to the respective build, which is expected to be available in the same namespace.
+```
+$ cat <<EOF | oc create -f -
+apiVersion: shipwright.io/v1beta1
+kind: BuildRun
+metadata:
+  name: buildah-golang-buildrun
```

Comment:
I think we can use `generatedName` - this will add a random suffix to the name:

```suggestion
  generatedName: buildah-golang-build-
```

Reply:
I think, the `generatedName` field when passed in the yaml doesn't work out. When I execute [1], it throws and error [0]: 
[0]
```
Name: "", Namespace: "default"
from server for: "STDIN": resource name may not be empty
```

[1]
```
$ oc apply -f - <<EOF                                                                                             [±BUILD-748 ●]
apiVersion: shipwright.io/v1beta1
kind: BuildRun
metadata:
  generatedName: s2i-nodejs-build-
spec:
  build:
    name: s2i-nodejs-build
EOF
```